content/en/blog/2025/autonomus_testing_with_antithesis.md

```diff
@@ -0,0 +1,121 @@
+---
+title: Autonomous Testing of etcd's Robustness
+author: "Marek Siarkowicz (Google)"
+draft: true
+---
+
+As a critical component of many production systems, including Kubernetes, the etcd project's first priority is reliability.
+Ensuring consistency and data safety requires our project contributors to continuously improve testing methodologies.
+In this article, we will describe how we used advanced simulation testing to uncover subtle bugs,
+validate the robustness of our releases, and increase our confidence in etcd's stability.
+We'll share our key findings and how they have improved etcd.
+
+## Enhancing etcd's Robustness Testing
+
+Many critical software systems depend on etcd to be correct and consistent, most notably as the primary datastore for Kubernetes.
+After some issues with the v3.5 release,
+the etcd maintainers developed a new [robustness testing framework](https://github.com/etcd-io/etcd/issues/14045)
+to better test for correctness under various failure scenarios. To further enhance our testing capabilities,
+we integrated an simulation testing platform from [Antithesis](https://antithesis.com/) into our workflow.
+
+The platform works by running the entire etcd cluster inside a deterministic hypervisor.
+This specialized environment gives the testing software complete control over every source of non-determinism,
+such as network behavior, thread scheduling, and system clocks.
+This means any bug it discovers can be perfectly and reliably reproduced.
+
+Within this simulated environment, the testing methodology shifts away from traditional, scenario-based tests.
+Instead of writing tests imperatively with strict assertions for one specific outcome, this approach uses declarative, property-based assertions about system behavior.
+These properties are high-level invariants about the system that must always hold true. For example,
+"data consistency is never violated" or "a watch event is never dropped."
+
+The platform then treats these properties not as passive checks, but as targets to break.
+It combines automated exploration with targeted fault injection,
+actively searching for the precise sequence of events and failures that will cause a property to be violated.
+This active search for violations is what allows the platform to uncover subtle bugs that result from complex combinations of factors.
+Antithesis refers to this approach as autonomous testing.
```

Comment:
```suggestion
Antithesis refers to this approach as "autonomous testing".
```

Reply:
Should either be quoted or should be capitalized (the latter if it's used as a term of art or a trademark)